resource/ui/ts/pages/graph.ts

```diff
@@ -22,23 +21,40 @@ module AdminViews {
        * displaying the same data set as retrieved by a QueryManager.
        */
       export module Page {
+          import property = _mithril.MithrilProperty;
           import metrics = Models.Metrics;
+
           class Controller implements _mithril.MithrilController {
-              manager:Utils.QueryCache<Models.Proto.QueryResultSet>;
+              manager:metrics.Executor;
+              axis:metrics.Axis;
               showRates:boolean;
               interval:number;
 
               timespan = metrics.time.Recent(10 * 60 * 1000);
-              sumquery = metrics.NewQuery(
-					  metrics.select.Avg("cr.node.calls.success.1").title("Successful calls"),
-					  metrics.select.Avg("cr.node.calls.error.1").title("Error calls")
+
+              // Define selectors.
+              private successCount = metrics.select.Avg("cr.node.calls.success.1")
+                .title("Successful calls");
+              private errorCount = metrics.select.Avg("cr.node.calls.error.1")
+                .title("Error calls");
+              private successRate = metrics.select.AvgRate("cr.node.calls.success.1")
+                .title("Successful call rate");
+              private errorRate = metrics.select.AvgRate("cr.node.calls.error.1")
+                .title("Error call rate");
+
+              // Define query.
+              private query = metrics.NewQuery(
+                      this.successCount, 
+                      this.errorCount, 
+                      this.successRate, 
+                      this.errorRate
 					)
```

Comment:
We need to have some consistency for spacing.  The close bracket at the end all the way to the right seems very strange to me.